src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaClientFetchFactory.java

```diff
@@ -840,6 +849,8 @@ else if (length != 0)
                 final int partitionId = partition.partitionId();
                 final int errorCode = partition.errorCode();
 
+                client.latestOffset = Math.max(client.latestOffset, partition.lastStableOffset());
+
```

Comment:
Under what conditions would `client.lastOffset` be greater than `partition.lastStableOffset()` ?